pkg/pxc/app/pmm.go

```diff
@@ -37,6 +42,31 @@ func PMMClient(cr *api.PerconaXtraDBCluster, spec *api.PMMSpec, secret *corev1.S
 			Name:  "CLIENT_PORT_MAX",
 			Value: "30105",
 		},
+		{
+			Name:  "CLUSTER_NAME",
+			Value: clusterName,
+		},
+		{
+			Name:  "PMM_AGENT_PRERUN_SCRIPT",
+			Value: "/var/lib/mysql/pmm-prerun.sh",
+		},
+		// PMM team added these flags which allows us to avoid
+		// container crash, but just restart pmm-agent till it recovers
+		// the connection.
+		{
+			Name:  "PMM_AGENT_SIDECAR",
+			Value: "true",
+		},
+		{
+			Name:  "PMM_AGENT_SIDECAR_SLEEP",
+			Value: "5",
+		},
+		// PMM team moved temp directory to /usr/local/percona/pmm2/tmp
+		// but it doesn't work on OpenShift so we set it back to /tmp
+		{
+			Name:  "PMM_AGENT_PATHS_TEMPDIR",
+			Value: "/tmp",
+		},
```

Comment:
are you sure that these changes won't cause cluster to be restarted after operator upgrade?

Reply:
You're right. I will check how to fix it.